examples/step-37/step-37.cc

```diff
@@ -721,30 +791,53 @@ namespace Step37
   // degree specified at the top of the file as well (otherwise, an exception
   // will be thrown at some point, since the computational kernel defined in
   // the templated LaplaceOperator class and the information from the finite
-  // element read out by MatrixFree will not match).
+  // element read out by MatrixFree will not match). The constructor of the
+  // triangulation needs to set an additional flag that tells the grid to
+  // conform to the 2:1 cell balance over vertices, which is needed for the
+  // convergence of the geometric multigrid routines. For the distributed
+  // grid, we also need to specifically enable the multigrid hierarchy.
   template <int dim>
   LaplaceProblem<dim>::LaplaceProblem ()
     :
+#ifdef DEAL_II_WITH_P4EST
+    triangulation(MPI_COMM_WORLD,
+                  Triangulation<dim>::limit_level_difference_at_vertices,
+                  parallel::distributed::Triangulation<dim>::construct_multigrid_hierarchy),
+#else
     triangulation (Triangulation<dim>::limit_level_difference_at_vertices),
+#endif
     fe (degree_finite_element),
     dof_handler (triangulation),
-    time_details (std::cout, false)
+    pcout (std::cout, Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0),
+    time_details (std::cout, false &&
```

Comment:
maybe mention it right after this line. When you just look at the code the fact that there is `false &&...` is unusual.